src/modelinstance.cpp

```diff
@@ -351,16 +351,47 @@ static Status applyLayoutConfiguration(const ModelConfig& config, std::shared_pt
     return StatusCode::OK;
 }
 
-const std::string RT_INFO_KEY{"model_info"};
+ov::AnyMap ModelInstance::getRTInfo(std::vector<std::string> path) {
+    ov::AnyMap anyMap, rtMap;
+    try {
+        rtMap = model->get_rt_info<ov::AnyMap>(path);
+    } catch (const ov::Exception& e) {
+        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Failed to get RTInfo; error:{}", e.what());
+        return anyMap;
+    } catch (...) {
+        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Failed to get RTInfo for path; unknown error");
+        return anyMap;
+    }
+    std::string type_name("St10shared_ptrIN2ov4MetaEE");
+    for (const auto& [key, value] : rtMap) {
+        if (*value.type_info().name() == *type_name.c_str()) {
```

Comment:
How does it work? This way you dereference only first letter of const char array, so you compare **S**. Shouldn't you use `std::strcmp(value.type_info().name(), typename.c_str()) == 0` or `std::string(value.type_info().name()) == type_name`?